core/Test/Tasty/Providers.hs

```diff
@@ -37,4 +40,13 @@ testFailed desc = Result
   , resultDescription = desc
   , resultShortDescription = "FAIL"
   , resultTime = 0
+  , printResultDetails = Nothing
   }
+
+-- | 'Result' of a failed test with custom details printer
+testFailedDetails
+  :: String               -- ^ description
+  -> ResultDetailsPrinter -- ^ details printer
+  -> Result
+testFailedDetails desc printer = (testFailed desc)
+  { printResultDetails = Just printer }
```

Comment:
I don't suppose we could sneak in `pure` or `return` here instead of `Just`?

Reply:
I wanted to have this as close to what upstream does as possible. Hence I used the `Just`. As I now removed the maybe the just is now redundant and we have one less thing to align for with upstream.